### PR TITLE
refs #1488 - improvements in wait* methods

### PR DESCRIPTION
**Pull request by wskorodecki**
Allowing 0 as a timeout value.
Added missing checks.
Issue: #1488

### CODE REVIEW

File: modules/casper.js [hunk 1]
```diff
@@ -2185,6 +2192,13 @@ Casper.prototype.waitForAlert = function(then, onTimeout, timeout) {
...
 Casper.prototype.waitForPopup = function waitForPopup(urlPattern, then, onTimeout, timeout) {
     "use strict";
+    if (then && !utils.isFunction(then)) {
+        throw new CasperError("waitForPopup() next step definition must be a function");
+    }
+    timeout = ~~timeout;
+    if (timeout < 0) {
+        throw new CasperError("waitForPopup() only accepts an integer >= 0 as a timeout value");
+    }
     return this.waitFor(function() {
```
Comment (istr) [inline modules/casper.js]: Please put this block of code in a helper function and call it rather than copying it around.

**Comment by istr** [comment]
Very useful. :+1: to merge this. As this is basically internal (refactoring) only, I do not see the need to update tests and/or documentation here. @mickaelandrieu @BIGjuevos

File: modules/casper.js [hunk 2]
```diff
@@ -2040,6 +2040,36 @@ Casper.prototype.warn = function warn(message) {
...
+        var isFunction = utils.isFunction(then); // Optimization to perform "isFunction" check only once.
+
+        if (isThenRequired && !isFunction) {
+            throw new CasperError(methodName + "() needs a step function");
+        } else if (then && !isFunction) {
+            throw new CasperError(methodName + "() next step definition must be a function");
+        }
+    }
+
+    timeout = ~~timeout || ~~defaultTimeout;
```
Comment (mickaelandrieu) [inline modules/casper.js]: shouldn't be ``var timeout = ~~timeout || ~~defaultTimeout;`` ?

**Comment by mickaelandrieu** [comment]
Ok to me too :) thank you @wskorodecki  and @istr for review

**Pull request merged**